DistroLauncher-Tests/DistroLauncher-Tests/PatchConfigTests.cpp

```diff
@@ -172,4 +179,54 @@ options=metadata
         std::string content(std::istreambuf_iterator<char>{result}, std::istreambuf_iterator<char>{});
         EXPECT_EQ(content, sampleContents(SampleStrings::wslConfAppend));
     }
+
+    /* Patching functions tests - asserts their behavior */
+
+    TEST(PatchingFn, CloudImgLabel)
+    {
+        // Makes the /etc/fstab exactly like 18.04's
+        std::istringstream input(sampleContents(SampleStrings::fstab1804));
+        std::stringstream output;
+        PatchingFunctions::RemoveCloudImgLabel(input, output);
+        // the patch function should have removed the only line the file contained.
+        EXPECT_EQ(output.str().size(), 0);
+    }
+    TEST(PatchingFn, CloudImgLabel2)
+    {
+        std::string slightlyChanged{sampleContents(SampleStrings::confComment)};
+        slightlyChanged.append("\n    ");
+        slightlyChanged.append(sampleContents(SampleStrings::fstab1804));
+        std::istringstream input(slightlyChanged);
+        std::stringstream output;
+
+        // Apply
+        PatchingFunctions::RemoveCloudImgLabel(input, output);
+
+        // Assert
+        // the patch function should have preserved the other line, which is just a comment.
+        EXPECT_EQ(output.str(), sampleContents(SampleStrings::confComment));
+    }
```

Comment:
Missing tests:
- fstab has leading whitespace (since you implemented behaviour to consider this).
- fstab does not contain the fstab1804 line

Reply:
```cpp
        std::string slightlyChanged{sampleContents(SampleStrings::confComment)};
        slightlyChanged.append("\n    ");
        slightlyChanged.append(sampleContents(SampleStrings::fstab1804));
```

That was causing the second line to start with spaces. But I'll clarify that test case.